### PR TITLE
Handle `ArraySection` in ``array_op.cpp`` ASR pass

### PR DESCRIPTION
**Pull request by czgdp1807**
Work in progress. Not ready for review.

### CODE REVIEW

**Comment by czgdp1807** [comment]
So this is ready for review. But before merging I would test it with https://github.com/lfortran/lfortran/pull/1630 to make sure that `minpack`'s `test_hybrd` compiles and runs correctly. Then we will merge this, then update https://github.com/lfortran/lfortran/pull/1630 and then finally merge it as well.

File: integration_tests/operator_overloading_05_module3.f90 [hunk 1]
```diff
@@ -8,12 +8,12 @@ pure subroutine insert_head( array )
...
-        tmp = array(0)
+        ! tmp = array(0)
         find_hole: do i=1, size(array, kind=4)-1
-            if ( array(i) >= tmp ) exit find_hole
-            array(i-1) = array(i)
+            ! if ( array(i) >= tmp ) exit find_hole
+            ! array(i-1) = array(i)
         end do find_hole
-        array(i-1) = tmp
+        ! array(i-1) = tmp
```
Comment (certik) [inline integration_tests/operator_overloading_05_module3.f90:16]: What is the plan here? The test doesn't work anymore?

File: src/libasr/pass/pass_manager.h [hunk 2]
```diff
@@ -195,7 +195,7 @@ namespace LCompilers {
                 "class_constructor",
                 "implied_do_loops",
                 "pass_list_expr",
-                "arr_slice",
+                // "arr_slice",
```
Comment (certik) [inline src/libasr/pass/pass_manager.h:198]: What about this --- intended?

**Comment by certik** [comment]
I think the PR looks ok. Does this implement any new feature? Are there tests that now pass? Or is this just refactoring?

File: src/libasr/pass/pass_manager.h (hunk 2 above)
Comment (czgdp1807) [reply src/libasr/pass/pass_manager.h:198]: It is intended. I have just commented it for now. Will remove this pass altogether in a separate PR.

Comment (certik) [reply src/libasr/pass/pass_manager.h:198]: Ok!

**Comment by czgdp1807** [comment]
> Are there tests that now pass? Or is this just refactoring?

This is for compiling `minpack`. See my comments in https://github.com/lfortran/lfortran/pull/1630#issuecomment-1531140265. `test_hybrd.f90` is segfaulting in https://github.com/lfortran/lfortran/pull/1630. After combining this PR and https://github.com/lfortran/lfortran/pull/1630 in https://github.com/lfortran/lfortran/pull/1654, `test_hybrd.f90` starts to compile and run but fails due to some numerical instability. Even GFortran fails due to minor change in how `minpack` is implementing `vecfcn` in `test_hybrd.f90`. See my comment https://github.com/lfortran/lfortran/pull/1654#issuecomment-1549334087 for more details.

**Review by certik — APPROVED** [review]
This looks good. Thanks!

**Pull request merged**